flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/Fabric8FlinkKubeClient.java

```diff
@@ -85,7 +85,15 @@ public Fabric8FlinkKubeClient(
             Configuration flinkConfig,
             NamespacedKubernetesClient client,
             ExecutorService executorService) {
-        this.clusterId = checkNotNull(flinkConfig.getString(KubernetesConfigOptions.CLUSTER_ID));
+        this.clusterId =
```

Comment:
why not just use `checkNotNull(flinkConfig.getString(KubernetesConfigOptions.CLUSTER_ID), String.format(
                                                        "Configuration option '%s' is not set.",
                                                        KubernetesConfigOptions.CLUSTER_ID.key())` ?

Reply:
`NullPointerException` didn't really feel suitable here as this is basically a misconfiguration on the user side.

NPE is more suitable for API misuse (something that can be basically avoided doing static code analysis).

Reply:
Get it. +1 for this.